pkg/ocm/provider/authorizer/open/open.go

```diff
@@ -67,6 +67,19 @@ func (a *authorizer) GetInfoByDomain(ctx context.Context, domain string) (*ocmpr
 		}
 	}
 
+	// there is a possibility that domain doesn't contain a scheme
+	// for example resource={"domain":"reva.docker"} and the call
+	// to ocmClient.Discover would fail.
+
+	// convert domain into lowercase to avoid adding scheme if the domain
+	// looks like this: HtTpS://domain.docker
+	domain = strings.ToLower(domain)
+
+	// add a scheme to domain if it doesn't have one.
+	if !strings.HasPrefix(domain, "http://") && !strings.HasPrefix(domain, "https://") {
+		domain = "https://" + domain
+	}
+
 	// not yet known: try to discover the remote OCM endpoint
 	ocmClient := client.NewClient(time.Duration(10)*time.Second, true)
 	ocmCaps, err := ocmClient.Discover(ctx, domain)
```

Comment:
I think I see where the issue is: `Discover()` requires an endpoint, whereas `domain` is what it is - a domain to be used as such in the `ProviderInfo` structure below.

So what I would do here is something like:
```code
var host string
if !HasPrefix(...) {
   host := "https://" + domain
} else {
   host := domain
}
ocmClient.Discover(ctx, host)
```
(also, no need to deal with `ToLower`)

Reply:
Thanks! 

I removed `toLower` and also couldn't name the new variable `host` since it would conflict with the line:

`host, _ := url.Parse(ocmCaps.Endpoint)`

so I have changed it to `endpoint`.